Make NNAPI linear converter accept flex inputs

axitkhurana opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61025
* #61024
* #61023
* **#61022**
* #61021
* #59529
* #59434

Summary:

As title

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_linear

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480749](https://our.internmc.facebook.com/intern/diff/D29480749)

Commit:
Make NNAPI linear converter accept flex inputs

Summary:

As title

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_linear

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480749).

Commit:
Update on "Make NNAPI linear converter accept flex inputs"


Summary:

As title

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_linear

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480749](https://our.internmc.facebook.com/intern/diff/D29480749)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480749).

Commit:
Update on "Make NNAPI linear converter accept flex inputs"


Summary:

As title

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_linear

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480749](https://our.internmc.facebook.com/intern/diff/D29480749)

[ghstack-poisoned]

dreiss commented on test/test_nnapi.py:
```diff
@@ -450,6 +450,9 @@ def test_upsample_nearest2d(self):
         self.check(torch.nn.Linear(16, 32), torch.randn(2, 16))
+        self.check(
+            torch.nn.Linear(16, 32), torch.randn(2, 16),
+            convert_args=[torch.zeros(0, 0)])
```
I think this should be `zeros(0, 16)`.  We don't need to support flex dims on the second dimension, since that has to match the weight dimension, right?

Commit:
Update on "Make NNAPI linear converter accept flex inputs"


Summary:

As title

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_linear

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480749](https://our.internmc.facebook.com/intern/diff/D29480749)

[ghstack-poisoned]

Commit:
Update on "Make NNAPI linear converter accept flex inputs"


Summary:

As title

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_linear

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480749](https://our.internmc.facebook.com/intern/diff/D29480749)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480749).

Commit:
Update on "Make NNAPI linear converter accept flex inputs"


Summary:

As title

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_linear

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480749](https://our.internmc.facebook.com/intern/diff/D29480749)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480749).

Commit:
Update on "Make NNAPI linear converter accept flex inputs"


Summary:

As title

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_linear

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480749](https://our.internmc.facebook.com/intern/diff/D29480749)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480749).
